### PR TITLE
Added an util that resolves event type and can be used by collections EventList.

### PR DESCRIPTION
**Pull request by nikomakela**
TH-1138.

NOTE: The code does not work yet. It needs LinkedEvents v2 and event type work done before it can be continued. It also needs changes from proxy https://github.com/City-of-Helsinki/events-helsinki-api-proxy/pull/64 and cms City-of-Helsinki/events-helsinki-cms#23

### CODE REVIEW

File: src/domain/eventSearch/utils.tsx [hunk 1]
```diff
@@ -443,3 +446,10 @@ export const getSearchQuery = (filters: Filters): string => {
 
   return buildQueryFromObject(newFilters);
 };
+
+export const getEventTypeFromRouteUrl = (url: string): EventType => {
+  if (matchPath(url, { path: ROUTES.COURSES, exact: true, strict: false })) {
+    return 'course';
+  }
+  return 'event';
+};
```
Comment (Sartonon) [inline src/domain/eventSearch/utils.tsx:455]: Maybe object map and looping could be used here. Would be easier to 'all' in the future. for example:

```
const routeToEventType = {
  [ROUTES.COURSES]: 'coure;
  [ROUTES.EVENTS]: 'event';
};
```

**Review by Sartonon — COMMENTED** [review]
Just one suggestion for gwtting event type but not necessary

File: src/domain/eventSearch/utils.tsx (hunk 1 above)
Comment (nikomakela) [reply src/domain/eventSearch/utils.tsx]: Added some changes.

**Comment by nikomakela** [comment]
Here are the events listed in Kesätekemistä lapsille ja nuorille -collection: 
image

Here are the courses listed in Kesäleirit -(draft) collection: 
image

So the one solution now fits for both!

**Comment by nikomakela** [comment]
The tests fails because new environment variables are missing from review, staging and production environments!

File: src/domain/collection/eventList/EventList.tsx [hunk 2]
```diff
@@ -39,13 +44,13 @@ const EventList: React.FC<Props> = ({ collection }) => {
...
-      eventType: 'event',
+      eventType: eventType ?? 'event',
     });
-  }, [locale, searchParams]);
+  }, [eventType, locale, searchParams]);
 
   const { data: eventsData, fetchMore, loading } = useEventListQuery({
     notifyOnNetworkStatusChange: true,
-    skip: !eventListQuery,
+    skip: !eventListQuery || !eventType,
```
Comment (Sartonon) [inline src/domain/collection/eventList/EventList.tsx:53]: Maybe no need to invlude both  variables in this check?

**Review by Sartonon — APPROVED** [review]

File: src/domain/collection/eventList/EventList.tsx (hunk 2 above)
Comment (nikomakela) [reply src/domain/collection/eventList/EventList.tsx:53]: It's there because eventType can be undefined when the url is given, but it's a bad one.

**Pull request merged**